client/hub.go

```
@@ -139,55 +145,75 @@ func getProviderPath(name string, version string) (string, error) {
 	if err != nil {
 		return err
 	}
 	if !isRegistered {
-		return fmt.Errorf("provider %s is not registered at https://hub.cloudquery.io", name)
+		return fmt.Errorf("provider %s is not registered at https://hub.cloudquery.io", pluginName)
 	}
 
 	providerURL := ""
 	checksumsURL := ""
+
+	fullPluginName := getProviderPluginName(pluginName)
 	if version == "" || version == "latest" {
 		version = "latest"
-		providerURL = fmt.Sprintf("https://github.com/%s/cq-provider-%s/releases/latest/download/cq-provider-%s_%s_%s",
-			org, name, name, runtime.GOOS, runtime.GOARCH)
-		checksumsURL = fmt.Sprintf("https://github.com/%s/cq-provider-%s/releases/latest/download/checksums.txt",
-			org, name)
+		providerURL = fmt.Sprintf("https://github.com/%s/cq-provider-%s/releases/latest/download/%s", org, pluginName, fullPluginName)
+		checksumsURL = fmt.Sprintf("https://github.com/%s/cq-provider-%s/releases/latest/download/checksums.txt", org, pluginName)
 	} else {
-		providerURL = fmt.Sprintf("https://github.com/%s/cq-provider-%s/releases/download/%s/cq-provider-%s_%s_%s",
-			org, name, version, name, runtime.GOOS, runtime.GOARCH)
-		checksumsURL = fmt.Sprintf("https://github.com/%s/cq-provider-%s/releases/download/%s/checksums.txt",
-			org, name, version)
+		providerURL = fmt.Sprintf("https://github.com/%s/cq-provider-%s/releases/download/%s/%s", org, pluginName, version, fullPluginName)
+		checksumsURL = fmt.Sprintf("https://github.com/%s/cq-provider-%s/releases/download/%s/checksums.txt", org, pluginName, version)
 	}
 
 	workingDir, err := os.Getwd()
 	if err != nil {
 		return err
 	}
 
-	pluginDir := fmt.Sprintf("%s/.cq/providers/%s/%s", workingDir, org, name)
+	pluginDir := filepath.Join(workingDir, ".cq", "providers", org, pluginName)
 	err = os.MkdirAll(pluginDir, os.ModePerm)
 	if err != nil {
 		return err
 	}
-	pluginPath := fmt.Sprintf("%s/%s-%s-%s", pluginDir, version, runtime.GOOS, runtime.GOARCH)
+	pluginPath, _ := getProviderPath(pluginName, version)
```

why ignore this error here?

---

should be fixed in later refactor